core/src/main/java/org/mule/runtime/core/internal/routing/UntilSuccessfulRouter.java

```diff
@@ -274,10 +274,12 @@ private CoreEvent eventWithCurrentContextDeleted(CoreEvent event) {
       if (throwable instanceof MessagingException) {
         exceptionEvent = ((MessagingException) throwable).getEvent();
       }
+      // MessagingException cause is set to null in order to avoid RETRY:EXHAUSTED ErrorType replacement
+      // during ErrorType resolution (see ChainErrorHandlingUtils and MessagingExceptionResolver)
       return new MessagingException(exceptionEvent,
                                     new RetryPolicyExhaustedException(createStaticMessage(UNTIL_SUCCESSFUL_MSG_PREFIX,
                                                                                           cause.getMessage()),
-                                                                      cause, owner),
+                                                                      null, owner),
```

Comment:
should this be done internally by `RetryPolicyExhaustedException`? What other uses are there for it?

Reply:
Good point. I will check for the uses and make the change if possible.